examples/step-31/step-31.cc

```diff
@@ -2081,8 +2081,13 @@ namespace Step31
     temperature_solution = tmp[0];
     old_temperature_solution = tmp[1];
 
+    temperature_constraints.distribute(temperature_solution);
```

Comment:
We comment on what we are doing in this function quite detailed. Could you put a comment in the comment above saying that we are also enforcing constraints to make the interpolated solution conforming on the new mesh?